[plan][cli]: Update logger timestamp format to compact style

## Description

Update the logger in `python/agentize/server/__main__.py` to use a compact timestamp format `[yy-mm-dd-hh:mm:ss]` (e.g., `[26-01-10-14:30:47]`) instead of the current ISO format (`[2026-01-10T14:30:47]`).

This change improves log readability by reducing timestamp length while maintaining sortability and unambiguous time representation.

## Proposed Solution

### Consensus Summary

We will apply the minimal, direct format change in the server logger and update the documented example to match, keeping scope tight. The plan uses a single-bracket log prefix and a 24-hour time with colons (`%y-%m-%d-%H:%M:%S`).

**Success criteria:**
- Server log lines emitted by `python/agentize/server/__main__.py` use a single-bracketed timestamp in the `%y-%m-%d-%H:%M:%S` format (e.g., `[26-01-10-14:30:47]`).
- `docs/feat/server.md` troubleshooting example reflects the new timestamp format.

**Out of scope:**
- Codebase-wide logging standardization across other modules
- Configurable timestamp formats per environment

### Codebase Analysis

**File changes:**

| File | Level | Purpose |
|------|-------|---------|
| `python/agentize/server/__main__.py` | minor | Update server log timestamp format string |
| `docs/feat/server.md` | minor | Update troubleshooting example timestamp |

### Interface Design

**Modified interfaces:**
- Log output timestamp format for server logs.

```diff
-    timestamp = datetime.now().strftime("%Y-%m-%dT%H:%M:%S")
+    timestamp = datetime.now().strftime("%y-%m-%d-%H:%M:%S")
```

### Implementation Steps

**Step 1: Documentation change** (Estimated: 1 LOC)
- File changes: `docs/feat/server.md`

```diff
- [2026-01-09T12:30:47] [ERROR] [__main__.py:163:query_project_items] GraphQL query failed: ...
+ [26-01-09-12:30:47] [ERROR] [__main__.py:163:query_project_items] GraphQL query failed: ...
```

**Step 2: Implementation change** (Estimated: 1 LOC)
- File changes: `python/agentize/server/__main__.py`

```diff
-    timestamp = datetime.now().strftime("%Y-%m-%dT%H:%M:%S")
+    timestamp = datetime.now().strftime("%y-%m-%d-%H:%M:%S")
```

**Total estimated complexity:** 2 LOC (Trivial)

### Success Criteria

- [ ] Server log prefix uses a single-bracketed `%y-%m-%d-%H:%M:%S` timestamp.
- [ ] `docs/feat/server.md` troubleshooting example matches the new format.
- [ ] No changes to documented command interfaces in `docs/feat/server.md`.

### Risks and Mitigations

| Risk | Likelihood | Impact | Mitigation |
|------|------------|--------|------------|
| Ambiguous format interpretation | M | M | Default to `%y-%m-%d-%H:%M:%S` (24-hour, colons for time) |
| Inconsistent formats across other loggers | M | L | Keep scope minimal; track as follow-up |
| Reduced compatibility with ISO parsers | L | L | Document format if log aggregation needed later |

## Related PR

TBD - will be updated when PR is created

--- a/docs/feat/server.md
+++ b/docs/feat/server.md
@@ -99,7 +99,7 @@ If `gh issue list` fails (e.g., network error, auth issue), the server returns a
 Error messages include source location (file:line:function) for quick debugging:
 
 ```
-[2026-01-09T12:30:47] [ERROR] [__main__.py:163:query_issue_project_status] GraphQL query failed: ...
+[26-01-09-12:30:47] [ERROR] [__main__.py:163:query_issue_project_status] GraphQL query failed: ...
 ```
 
 For additional context (query and variables), set `HANDSOFF_DEBUG=1`:

--- a/python/agentize/server/__main__.py
+++ b/python/agentize/server/__main__.py
@@ -33,7 +33,7 @@ def _log(msg: str, level: str = "INFO") -> None:
     filename = os.path.basename(frame.f_code.co_filename)
     lineno = frame.f_lineno
     func = frame.f_code.co_name
-    timestamp = datetime.now().strftime("%Y-%m-%dT%H:%M:%S")
+    timestamp = datetime.now().strftime("%y-%m-%d-%H:%M:%S")
 
     output = f"[{timestamp}] [{level}] [{filename}:{lineno}:{func}] {msg}"
     print(output, file=sys.stderr if level == "ERROR" else sys.stdout)

